aiotg/bot.py

```diff
@@ -78,6 +82,42 @@ def no_handle(mt):
             )
             self._process_updates(updates)
 
+    def webhook_loop(self, loop):
+        """
+        Starts aiohttp web server.
+        """
+        app = web.Application(loop=loop)
+
+        async def handle(request):
+            update = await request.json()
+            self._process_update(update)
+            return web.Response()
+        app.router.add_route('POST', urlparse(self.webhook_url).path, handle)
+        web.run_app(app)
+
+    def set_webhook(self, url, **options):
+        """
+        Register you webhook url for Telegram service.
+
+        :Example:
+
+        >>> bot.set_webhook(url='https://yourserver.com/webhook/<token>')
+
+        Additional documentation on https://core.telegram.org/bots/api#setwebhook
+        """
+        self.webhook_url = url
+        return asyncio.get_event_loop().run_until_complete(self.api_call(
+            'setWebhook',
+            url=url,
+            **options
+        ))
+
+    def stop_webhook(self):
+        """
+        Unregister webhook url. Activates getUpdates
+        """
+        self.set_webhook(url="")
+
     def run(self):
```

Comment:
we can change it to run(self, webhook=None) or webhook_url, i'm never sure about the naming :)